src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -428,11 +430,22 @@ private async Task<bool> NewerManifestPackageExists(ManifestId manifest)
 
         private IEnumerable<(ManifestId, ManifestVersion)> ParseRollbackDefinitionFile(string rollbackDefinitionFilePath)
         {
-            if (!File.Exists(rollbackDefinitionFilePath))
+            string fileContent;
+            if (File.Exists(rollbackDefinitionFilePath))
             {
-                throw new ArgumentException(string.Format(LocalizableStrings.RollbackDefinitionFileDoesNotExist, rollbackDefinitionFilePath));
+                fileContent = File.ReadAllText(rollbackDefinitionFilePath);
+            }
+            else
+            {
+                try
+                {
+                    fileContent = (new HttpClient()).GetStringAsync(rollbackDefinitionFilePath).Result;
```

Comment:
I think it would be better to try to check if the string looks like a URL (perhaps by checking if it starts with `http://` or `https://`, though it would be good to get guidance from someone who knows more) rather than just checking whether the file exists and then indiscriminately trying to download it.

Also, what does the error experience look like if the URI doesn't exist, or can't be reached?  I think we would want a somewhat "graceful" exception in that case, ie not printing the stack trace.

Related to that, we might want to print a message that we're trying to download the file before we do it, so that if it takes a while there is some feedback.

Reply:
Take a look at https://docs.microsoft.com/en-us/dotnet/api/system.uri.trycreate?view=net-5.0. I think you can create a URI, then verify the result's URI scheme to be HTTPS - I don't think we should allow for HTTP as that is not secure and we shouldn't trust non-HTTPS.

You could maybe try sending a HEAD request to the URL, but what if you're offline or restricted to an intranet?

Reply:
Sure, I've converted to check if it can be parsed as a uri using the api @joeloff linked. When the URI doesn't exist or can't be reached we handle the resulting exception elsewhere in the code so it gets formatted before being displayed. 
I'm not adding a message before downloading the file for now since we're past the loc deadline.